Fixed SignInUp Modal misalignment for devices smaller than 400px width

ehconitin opened the pull request:
Hi @Bonapara,
Issue #6385 

I encountered an issue with the Modal component where its width was fixed at 400px. While the container housing the Modal adjusted its size based on the screen width, the Modal itself remained at 400px regardless of the screen size.

I have implemented a change to address this problem. Could you please review the changes and let me know your thoughts?

Thank you!



https://github.com/user-attachments/assets/8358aacb-d6c3-440e-895e-7abc4f8a3534

ehconitin commented:
@Bonapara 
Addressed issue #5740 

for devices width viewport less than 768px(mobile) will have onboarding screen fullscreened.
Btw I noticed one more bug, pricing screen does not go into dark mode.

https://github.com/user-attachments/assets/789fec7e-e0bc-424e-9b57-c7e06b10abd4


https://github.com/user-attachments/assets/80f0d3d1-6602-4415-8e35-1f1945d9997b

Bonapara commented:
Hi @ehconitin, thanks so much for this PR. Do you think you can address the issue of pricing not switching to dark mode in the same PR?

ehconitin commented:
@Bonapara 
I did look into it a bit.
I think that needs further digging.
I will get back to you if I figure it out.

ehconitin commented:
@Bonapara 
I made changes to ensure consistent theme management just like we discussed.
I would like to know if this a good approach, if yes I will cleanup the code a bit for merge.

ehconitin commented:
I had a look at PR#5758

Creating variants should be easy.
Ill do that .
Right now the deletion modal on mobile screen goes fullscreen and not aligned.
Will fix it.

ehconitin commented:
Commited a temporay fix for modals.
This works.
But I think Modal overhaul is required.
It is being used in many places with atleast 3 different wrappers I could find in short time.

ehconitin commented:
@FelixMalfait @Bonapara

I have created a new EnhancedModalLayout component that integrates the closable functionality, thereby consolidating the basic layout and closing properties into a single component. This change eliminates the need for two separate components, ModalLayout and Modal, for these functionalities.

Additionally, I introduced modalVariants as requested in [this comment](https://github.com/twentyhq/twenty/pull/5758#issuecomment-2152809841).

Could you please confirm if the spreadsheet-import modal needs to be hotkey-scoped? For now, I have added it. I have reviewed all modals to ensure the behavior is consistent, but please let me know if I missed anything.

Bonapara commented:
@lucasbordeau, maybe you have some thoughts too!

FelixMalfait commented on packages/twenty-front/src/modules/ui/input/color-scheme/components/ColorSchemePicker.tsx:
```diff
@@ -35,31 +35,37 @@ export const ColorSchemePicker = ({
-);
+}: ColorSchemePickerProps) => {
+  const handleOnChange = (ColorScheme: ColorScheme) => {
+    localStorage.setItem('app-theme', ColorScheme);
```
Not a fan of introducing this, I think there are more elegant ways to solve this :)
Would you mind reverting the changes related to that? We can always solve them in another PR/issue and take time to do it properly (sorry I know this kind of contradict what was said before)

ehconitin replied:
Got it!
I just wanted to point out the theme inconsistencies.
Will revert it.
And yes this could be solved in much more elegant way, I was trying to figure out how this works.
Thanks

FelixMalfait commented on packages/twenty-front/src/modules/ui/layout/modal/components/EnhancedModalLayout.tsx:
```diff
@@ -0,0 +1,252 @@
+};
+
+// This component should be used over Modal when seeking a modal feel without modal state (hotkeyScope etc)
+export const EnhancedModalLayout = ({
```
I'll review the PR more in-depth to better understand what you did, but at first look, it isn't very clear what the intention is here. You named the component "EnhancedModalLayout" which is quite generic and not descriptive enough (proof if you were forced to add a comment, if you have a clean API for your component and a clean name, then its purpose should be obvious)

ehconitin replied:
Oops, that comment's a mistake.

The point of introducing the new `EnhancedModalLayout` is to get rid of having two different components—one for the modal `ModalLayout` and another for hotscoped `Modal`. Once this `EnhancedModalLayout` is added, we won’t need the existing `ModalLayout` or `Modal`.

After that, we can rename it to either Modal or ModalLayout, whichever fits better. I just called it EnhancedModalLayout for now to keep things clear.

FelixMalfait commented on packages/twenty-front/src/modules/ui/layout/modal/components/EnhancedModalLayout.tsx:
```diff
@@ -0,0 +1,252 @@
+  className,
+  hotkeyScope = ModalHotkeyScope.Default,
+  onEnter,
+  isClosable = false,
```
You say in the comment above the goal was to create a modal without hotkey scope, but you ended having "isClosable", "hotkeyScope" etc so I don't really understand the goal. Will keep reading more to understand

ehconitin replied:
That comment is a mistake, mb :(

FelixMalfait commented on packages/twenty-ui/src/theme/constants/BackgroundDark.ts:
```diff
@@ -24,6 +24,7 @@ export const BACKGROUND_DARK = {
     danger: RGBA(COLOR.red, 0.08),
   },
   overlay: RGBA(GRAY_SCALE.gray80, 0.8),
+  overlay2: RGBA(GRAY_SCALE.gray80, 0.4),
```
overlayPrimary, overlaySecondary?

ehconitin replied:
yep, will make the change
This was just lazy on my part

FelixMalfait commented on packages/twenty-ui/src/theme/constants/BackgroundLight.ts:
```diff
@@ -24,6 +24,7 @@ export const BACKGROUND_LIGHT = {
     danger: RGBA(COLOR.red, 0.08),
   },
   overlay: RGBA(GRAY_SCALE.gray80, 0.8),
+  overlay2: RGBA(GRAY_SCALE.gray80, 0.4),
```
overlayPrimary, overlaySecondary?

FelixMalfait commented on packages/twenty-front/src/modules/ui/layout/modal/components/EnhancedModalLayout.tsx:
```diff
@@ -0,0 +1,252 @@
+      <StyledModalDiv
+        // framer-motion seems to have typing problems with refs
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
```
We try to avoid @ts-ignore, and in that case I don't have any error on my side?
<img width="171" alt="Screenshot 2024-08-07 at 09 42 44" src="https://github.com/user-attachments/assets/caf9b8af-1428-442a-81c3-6c6eb561e28d">

If you use VSCode you can try cmd+shift+p -> "Reload window", sometimes it helps when Typescript says something wrong

ehconitin replied:
okay, Got it!

FelixMalfait commented on packages/twenty-front/src/modules/ui/layout/modal/components/ModalLayout.tsx:
```diff
@@ -1,21 +1,26 @@
-import React from 'react';
```
I have a feeling we don't need to have both `ModalLayout` and `EnhancedModalLayout`, or that I didn't really get your intention. If we think we really do need to have 2 components, then we need to create a story for that new modal component

ehconitin replied:
No we dont, thats the whole point.
I am sorry if I was unclear in my comments or description.
I integrated ModalLayout and Modal into EhnhancedModelLayout (naming could be way better)

FelixMalfait commented on packages/twenty-front/src/modules/ui/layout/modal/components/__stories__/ModalLayout.stories.tsx:
```diff
@@ -1,31 +1,31 @@
 
 export default meta;
-type Story = StoryObj<typeof ModalLayout>;
+type Story = StoryObj<typeof EnhancedModalLayout>;
```
Oh so we deprecate ModalLayout? We not leave it named as ModalLayout and leave it like this then?

I saw in your nice PR you commented some code. Don't be afraid to delete files and code, that's what Git is for, we can always revert if it wasn't the right strategy!

ehconitin replied:
yep we decpreciate ModalLayout and Modal(which is used for hotscoped)

FelixMalfait replied:
Sorry I misunderstood!

FelixMalfait replied:
Yes I agree, we haven't finished migrating components to twenty-ui! It's on the roadmap but I didn't really follow this topic closely

FelixMalfait reviewed (commented):
Will be a nice improvement! I think I didn't fully get the intention/strategy behind that new component, hopefully we can clarify and merge this. Thanks (again and again!) for your work

ehconitin commented:
@FelixMalfait 

This is how the code would look after changes.
Could you please review once more :)
Please let me know if I missed something.

FelixMalfait commented on packages/twenty-front/src/modules/spreadsheet-import/components/ModalWrapper.tsx:
```diff
@@ -43,8 +43,6 @@ export const ModalWrapper = ({
     <>
       {isOpen && (
         <StyledModal size="large" onClose={onClose} isClosable={true}>
-          {/*Remove onClose and isClosable if do not this compoenent key hotscoped */}
```
Yes, the behavior seems acceptable like that to me

ehconitin replied:
wait, so should be hotscoped or not?
I am sorry but Just want to confirm, cuz accidental touch might lead to loss of all that import steps user might be doing.
Presently it is hotscoped which it initially was not.
